renderer/components/Pay/PayAddressField.js

```diff
@@ -0,0 +1,101 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import { Box } from 'rebass'
+import { Keyframes } from 'react-spring/renderprops.cjs'
+import { intlShape } from '@zap/i18n'
+import { LightningInvoiceInput } from 'components/Form'
+import messages from './messages'
+import { PAY_FORM_STEPS } from './constants'
+
+/**
+ * Animation to handle showing/hiding the payReq field.
+ */
+const ShowHidePayReq = Keyframes.Spring({
+  small: { height: 48 },
+  big: async (next, cancel, ownProps) => {
+    ownProps.context.focusPayReqInput()
+    await next({ height: 110, immediate: true })
+  },
+})
+
+class PayAddressField extends React.Component {
+  static propTypes = {
+    chain: PropTypes.string.isRequired,
+    currentStep: PropTypes.string.isRequired,
+    handlePayReqChange: PropTypes.func.isRequired,
+    intl: intlShape.isRequired,
+    isLn: PropTypes.bool,
+    network: PropTypes.string.isRequired,
+    redirectPayReq: PropTypes.object,
+  }
+
+  static defaultProps = {
+    redirectPayReq: null,
+  }
+
+  payReqInput = React.createRef()
+
+  /**
+   * focusPayReqInput - Focus the payReq input.
+   */
+  focusPayReqInput = () => {
+    if (this.payReqInput.current) {
+      this.payReqInput.current.focus()
+    }
+  }
+
+  render() {
+    const {
+      chain,
+      currentStep,
+      handlePayReqChange,
+      intl,
+      isLn,
+      network,
+      redirectPayReq,
+    } = this.props
+
+    const payReq_label =
+      currentStep === PAY_FORM_STEPS.address
```

Comment:
Can we re-write this in a more understandable way? These nested ternary operators look confusing

Reply:
will do

Reply:
also please rename `payReq_label ` to a camel case `payReqLabel`